website/docs/for-ios/how-to/define-api-in-typescript.md

```diff
@@ -8,34 +8,189 @@ import TabItem from '@theme/TabItem';
+    self.onPortalLoaded = onPortalLoaded
+    super.init()
+  }
+
+  @objc func portalLoaded(_ call: CAPPluginCall) {
+    onPortalLoaded()
+    call.resolve()
+  }
+}
+```
+
+:::info
+You'll also need a file called `PortalLoadedPlugin.m` to create Objective-C bindings through helper macros that Capacitor provides.
+Below is an example `PortalLoadedPlugin` Objective-C bindings:
+
+```c title=PortalLoadedPlugin.m
+#import <Capacitor/Capacitor.h>
+
+CAP_PLUGIN(PortalLoadedPlugin, "PortalLoadedPlugin",
+  CAP_PLUGIN_METHOD(portalLoaded, CAPPluginReturnNone);
+)
+```
+
+:::
+
 
-We strongly recommend using TypeScript to create a type defintion file which can be used to define your API. This way, there is a central source of truth for the API across Android and iOS as well as having type defintions for the web code.
+### Using the PortalLoadedPlugin
+
+To use the `PortalLoadedPlugin` class, we need to create an instance of it and add it to the portal to be rendered. In our example, we'll fade
+the view in over half a second to ease the transition so we don't get an initial page flash when the portal is rendered.
+
+```swift title=PortalViewController.swift
+import UIKit
+import IonicPortals
+
+class PortalViewController: UIViewController {
+  // Create an instance of PortalLoadedPlugin to fade-in the 
+  // portal when our method has been called.
+  lazy var loadedPlugin = PortalLoadedPlugin { [weak self] in 
+    DispatchQueue.main.async {
+      UIView.animate(withDuration: 0.5) {
+        self?.portalView.alpha = 1.0
+      }
+    }
+  }()
+
+  // Create the PortalUIView by passing in our portal with the 
+  // instance of `PortalLoadedPlugin` added
+  lazy var portalView: PortalUIView = {
+    let portalView = PortalUIView(
+      portal: Portal(name: "foo")
+        .adding(loadedPlugin)
+    )
+
+    portalView.alpha = 0
+    return portalView
+  }()
+
+  // Add portalView to the view hierarchy
+  override func viewDidLoad() {
+    super.viewDidLoad()
+    portalView.translatesAutoresizingMasksIntoConstraints = false
+    view.addSubview(portalView)
+    NSLayoutConstraint.activate([
+      portalView.topAnchor.constraint(equalTo: view.topAnchor),
+      portalView.trailingAnchor.constraint(equalTo: view.trailingAnchor),
+      portalView.bottomAnchor.constraint(equalTo: view.bottomAnchor)
+      portalView.leadingAnchor.constraint(equalTo: view.leadingAnchor),
+    ])
+  }
+}
+```
+
+### Calling the Plugin Code from the Web
+
+Once the Plugin has been defied, implemented, and initialized in the native code, it will also need to be registered on the web.
+
+```typescript title=PortalLoadedPlugin/PortalLoadedPlugin.ts
+import { registerPlugin } from '@capacitor/core';
+import { PortalLoadedPlugin } from '../definitions';
+
+const PortalLoaded = registerPlugin<PortalLoadedPlugin>('PortalLoaded', {
+ // We create a no-op on the web for development purposes. This no-op plugin will only be loaded when running the plugin 
```

Comment:
It may be a good idea to document why we would opt to not export the Plugin directly to be used by the rest of the web application.

Reply:
can we add this before this doc is done? does it need to be elsewhere?